pallets/governance-origins/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019-2022 PureStake Inc.
```

Comment:
I am not too convinced of adding a pallet without storage nor extrinsics. What was the main problem this was trying to address?

Reply:
So the reason I am saying this is because we might want to have different origins/tracks for different runtimes, and this pallet makes it not that easy to customize for each runtime.

If the issue was to access the origins in the precompile, I think you can pass a new generic to `ReferendaPrecompile`:
`pub struct ReferendaPrecompile<Runtime, GovernanceOrigins: TryFrom<u8>>(PhantomData<(Runtime, GovernanceOrigins)>);`

Then you just define the appropriate traits (probably `OriginOf<Runtime>: From<GovernanceOrigins>`) and in the runtime you pass the previous `pallet_custom_origins::Origin` as a generic

Reply:
I'd be probably fine if the pallet was configurable for Origins and Tracks but that's not the case currently and I feel its work for another PR, not for this one. What do u think?

Reply:
> I think you can pass a new generic to `ReferendaPrecompile`:
`pub struct ReferendaPrecompile<Runtime, GovernanceOrigins: TryFrom<u8>>(PhantomData<(Runtime, GovernanceOrigins)>);`

Ok, will make this change, thanks for the suggestion!

> What was the main problem this was trying to address?

As you mentioned, to use the `Origins` in the referenda-precompile (as an input for `pallet_referenda::submit`), extracted `pallet_custom_origins` into its own pallet `pallet-governance-origins`.

> I'd be probably fine if the pallet was configurable for Origins and Tracks but that's not the case currently and I feel its work for another PR, not for this one. What do u think

Tracks are defined in the runtime (`$runtime/governance/tracks.rs`). It only defines the Origins, which I expected to be the same set for all runtimes. The spender limits are still configured in the runtime, but the set of Origins I expected to stay consistent...

Reply:
Yeah not sure to be honest, I was just thinking about potentially adding a track that dispatches with a new origin in Moonbase for testing purposes. But I probably do not have the entire context so as to how common this will be

Reply:
Followed this suggestion